packages/backend/src/vscode-youi-events.ts

```diff
@@ -176,21 +178,22 @@ export class VSCodeYouiEvents implements YouiEvents {
 
     if (success) {
       if (!isNil(targetFolderPath)) {
-        const targetFolderUri: vscode.Uri = vscode.Uri.file(targetFolderPath);
-
-        if (selectedWorkspace === this.messages.open_in_a_new_workspace) {
-          void vscode.commands.executeCommand("vscode.openFolder", targetFolderUri);
-        } else if (selectedWorkspace === this.messages.add_to_workspace) {
-          const wsFoldersQuantity = size(vscode.workspace.workspaceFolders);
-          vscode.workspace.updateWorkspaceFolders(wsFoldersQuantity, null, {
-            uri: targetFolderUri,
-          });
-          if (isNil(vscode.workspace.workspaceFile)) {
-            const workspaceFileUri = WorkspaceFile.create(targetFolderUri.fsPath);
-            void vscode.commands.executeCommand("vscode.openFolder", workspaceFileUri);
+        let targetFolderUri: vscode.Uri = vscode.Uri.file(targetFolderPath);
+        if(validator.isURL(targetFolderPath, IS_URL_DEFAULT_OPTIONS)){
+          if(selectedWorkspace === this.messages.add_to_workspace){
+            targetFolderUri = vscode.Uri.parse(targetFolderPath);
+            const createWsFunc = WorkspaceFile.createUri;
+            this.addOrCreateProjectWorkspace(targetFolderUri, createWsFunc);
           }
+          //TODO: add a popup error if the user not in add_to_workspace mode.
         }
+        else if (selectedWorkspace === this.messages.open_in_a_new_workspace) {
+          void vscode.commands.executeCommand("vscode.openFolder", targetFolderUri);
+        } else if (selectedWorkspace === this.messages.add_to_workspace) {
+          const createWsFunc = WorkspaceFile.create;
+          this.addOrCreateProjectWorkspace(targetFolderUri, createWsFunc);
       }
+    }
```

Comment:
consider splitting the flow for cases where `targetFolderPath` is represented as a URL (new) and as a string (old) to avoid making the code more complex